src/main/java/com/expedia/pitchfork/systems/datadog/forwarder/DatadogSpansDispatcher.java

```diff
@@ -116,8 +131,9 @@ private void flush(List<Span> spans) {
     public void addSpan(Span span) {
         try {
             this.pending.add(span);
-        } catch (Exception e) {
-            logger.error("operation=addSpan", e);
+        } catch (IllegalStateException e) {
+            failureCounter.increment();
+            logger.error("Unable to add span with id {} to queue. Queue full.", span.id());
```

Comment:
You're logging "queue full" as it's for sure the cause of the `IllegalStateException`. Just double checking, is it the only possible cause of such exception?

Reply:
Yeap, it is

![image](https://user-images.githubusercontent.com/1187103/121327721-9a91c600-c90b-11eb-940f-1afa85629878.png)